cli/src/pcluster/resources/login_node/user_data.sh

```diff
@@ -0,0 +1,233 @@
+      }
+
+      [ -f /etc/profile.d/proxy.sh ] && . /etc/profile.d/proxy.sh
+
+      # Configure AWS CLI using the expected overrides, if any.
+      [ -f /etc/profile.d/aws-cli-default-config.sh ] && . /etc/profile.d/aws-cli-default-config.sh
+
+      custom_cookbook=${CustomChefCookbook}
+      export _region=${AWS::Region}
+
+      s3_url=${AWS::URLSuffix}
+      if [ "${!custom_cookbook}" != "NONE" ]; then
+        if [[ "${!custom_cookbook}" =~ ^s3://([^/]*)(.*) ]]; then
+          # Set the socket connection timeout to 15s. Emperically, it seems like the actual
+          # timeout is 8x(cli-connect-timeout). i.e. if cli-connection-timeout is set to
+          # 60s, the call will timeout the connect attempt at 8m. Setting it to 15s, causes
+          # each attempt to take 240s, so 2m * 3 attempts will result in a failure after 6
+          # minutes.
+          S3API_RESULT=$(AWS_RETRY_MODE=standard aws s3api get-bucket-location --cli-connect-timeout 15 --bucket ${!BASH_REMATCH[1]} --region ${AWS::Region} 2>&1) || error_exit "${!S3API_RESULT}"
+          bucket_region=$(echo "${!S3API_RESULT}" | jq -r '.LocationConstraint')
+          if [[ "${!bucket_region}" == null ]]; then
+            bucket_region="us-east-1"
+          fi
+          cookbook_url=$(aws s3 presign "${!custom_cookbook}" --region "${!bucket_region}")
+        else
+          cookbook_url=${!custom_cookbook}
+        fi
+      fi
+      export PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/opt/aws/bin
+      export parallelcluster_version=aws-parallelcluster-${ParallelClusterVersion}
+      export cookbook_version=${CookbookVersion}
+      export chef_version=${ChefVersion}
+      export berkshelf_version=${BerkshelfVersion}
+      if [ -f /opt/parallelcluster/.bootstrapped ]; then
+        installed_version=$(cat /opt/parallelcluster/.bootstrapped)
+        if [ "${!cookbook_version}" != "${!installed_version}" ]; then
+          cookbook_version_number=$(echo ${!cookbook_version} | awk -F- '{print $NF}')
+          installed_version_number=$(echo ${!installed_version} | awk -F- '{print $NF}')
+          error_exit "This AMI was created with ParallelCluster ${!installed_version_number}, but is trying to be used with ParallelCluster ${!cookbook_version_number}. Please either use an AMI created with ParallelCluster ${!cookbook_version_number} or change your ParallelCluster to ${!installed_version_number}"
+        fi
+      else
+        error_exit "This AMI was not baked by ParallelCluster. Please use pcluster build-image command to create an AMI by providing your AMI as parent image."
+      fi
+      if [ "${!custom_cookbook}" != "NONE" ]; then
+        curl --retry 3 -v -L -o /etc/chef/aws-parallelcluster-cookbook.tgz ${!cookbook_url}
+        vendor_cookbook
+      fi
+      cd /tmp
+
+      mkdir -p /etc/chef/ohai/hints
+      touch /etc/chef/ohai/hints/ec2.json
+
+      # measure start time
+      start=$(date +%s)
```

Comment:
The `startup_time` metrics probably don't need to be published in this case. In any case the `start` and `end` variables are not being used.